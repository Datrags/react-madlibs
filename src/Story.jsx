//Story
//Displays the story from user inputed words
//Takes an object of words as props 
const Story = ({wordsObj}) => {
    //destructure the words object we recieve and place the words into the story
    const {name, adj1,adj2,adj3,adj4,adj5,plural_noun,verb_ing,noun,animal} = wordsObj;
    const theStory = `
    Once upon a time, there was a ${adj1} guy named ${name}.
    One day, ${name} decided to visit the ${adj2} zoo.
    At the Zoo, the first thing ${name} saw was a ${animal} eating ${plural_noun}.
    It was very ${adj3} to see that!
    Next, ${name} saw an ${adj4} ${animal} that was ${verb_ing} with a ${noun}. 
    ${name} laughed and said, "That's the most ${adj5} thing I've ever seen!"`;

    const dividedStory = theStory.split("\n").map((line, i) => <p key={"s"+i}>{line}</p>);

    return <div>{dividedStory}</div>
}

export default Story;