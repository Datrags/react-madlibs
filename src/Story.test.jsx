import React from "react";
import { render } from "@testing-library/react";
import Story from "./Story";

const storyData = {
    name: "Joe",
    adj1: "Fuzzy",
    adj2: "Nimble",
    animal: "Giraffe",
    plural_noun: "Chickens",
    adj3: "Fragrant",
    adj4: "Luminous",
    verb_ing: "Jumping",
    noun: "Nacho",
    adj5: "Zealous"
}

it("renders without crashing", function() {
  render(<Story wordsObj={storyData}/>);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Story wordsObj={storyData}/>);
  expect(asFragment()).toMatchSnapshot();
});

// it("renders a full story with data inputted", function() {
//     const story = render(<Story wordsObj={storyData}/>)
//     const giraffe = story.getByText(`Once upon a time, there was a Fuzzy guy named Joe.`);
//     expect(giraffe).toBeInTheDocument();
// })