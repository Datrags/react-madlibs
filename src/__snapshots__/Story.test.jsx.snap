// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Once upon a time, there was a Fuzzy guy named Joe.
    One day, Joe decided to visit the Nimble zoo.

    
    At the Zoo, the first thing Joe saw was a Giraffe eating Chickens.
    It was very Fragrant to see that!

    
    Next, Joe saw an Luminous Giraffe that was Jumping with a Nacho. 
    Joe laughed and said, "That's the most Zealous thing I've ever seen!"
  </p>
</DocumentFragment>
`;
